### PR TITLE
feat-mobile: add shimmer claiming

### PR DESCRIPTION
**Pull request by paul-boegelsack**
## Summary

Setup shimmer claiming for mobile onboarding

## Changelog

```
- Adds shimmer claiming router
- Adds reward and success view for claiming
- Adds a claiming list component for mobile
```

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
	- [ ] MacOS
	- [ ] Linux
	- [ ] Windows
- __Mobile__
	- [ ] iOS
	- [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
Im getting an error when I try to claim my rewards 
```
Error: Unable to initialize Shimmer claiming account
    at initialiseFirstShimmerClaimingAccount
```
line `throw new CannotInitialiseShimmerClaimingAccountError()`

**Comment by paul-boegelsack** [comment]
@Tuditi "Stronghold reported a procedure error: engine: fatal engine error: vault `VaultId(TcAMhLekJE6GhvWh6qTYL-T2DKgaUKz7)` does not exist" is thrown at initialiseFirstShimmerClaimingAccount.ts line 35.

**Comment by paul-boegelsack** [comment]
> Im getting an error when I try to claim my rewards
> 
> ```
> Error: Unable to initialize Shimmer claiming account
>     at initialiseFirstShimmerClaimingAccount
> ```
> 
> line `throw new CannotInitialiseShimmerClaimingAccountError()`

Should work now thanks to @Tuditi

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]
Have one tiny requested change - looks great!

**Comment by maxwellmattryan** [comment]
I'm getting this error when trying to claim in a scenario where I've got funds on account index 2, address index 5:
```
TypeError: Cannot read properties of undefined (reading 'toString')
    at getAccountAssetsForSelectedAccount (getAccountAssetsForSelectedAccount.ts:16:1)
    at selected-account-assets.store.ts:42:1
    at sync (index.mjs:77:1)
    at index.mjs:95:1
    at Object.subscribe (index.mjs:48:1)
    at subscribe (index.mjs:55:1)
    at get_store_value (index.mjs:60:1)
    at getOutputOptions (getOutputOptions.ts:24:1)
    at claimShimmerRewardsForShimmerClaimingAccount (claimShimmerRewards.ts:67:1)
    at async claimShimmerRewardsForShimmerClaimingAccounts (claimShimmerRewards.ts:42:1)
```

File: packages/mobile/views/onboarding/views/profile-setup/views/index.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
 export { default as EnterNameView } from './EnterNameView.svelte'
 export { default as SetupView } from './SetupView.svelte'
+export { default as SetupClaimedView } from './SetupClaimedView.svelte'
 export { default as SetupRecoveredView } from './SetupRecoveredView.svelte'
```
[the remark this answers is not in the document]
Comment (paul-boegelsack) [reply packages/mobile/views/onboarding/views/profile-setup/views/index.js]: Thank you for the catch 👍

**Comment by paul-boegelsack** [comment]
> I'm getting this error when trying to claim in a scenario where I've got funds on account index 2, address index 5:
> 
> ```
> TypeError: Cannot read properties of undefined (reading 'toString')
>     at getAccountAssetsForSelectedAccount (getAccountAssetsForSelectedAccount.ts:16:1)
>     at selected-account-assets.store.ts:42:1
>     at sync (index.mjs:77:1)
>     at index.mjs:95:1
>     at Object.subscribe (index.mjs:48:1)
>     at subscribe (index.mjs:55:1)
>     at get_store_value (index.mjs:60:1)
>     at getOutputOptions (getOutputOptions.ts:24:1)
>     at claimShimmerRewardsForShimmerClaimingAccount (claimShimmerRewards.ts:67:1)
>     at async claimShimmerRewardsForShimmerClaimingAccounts (claimShimmerRewards.ts:42:1)
> ```

I tried to reproduce but without success. I created this with your fund spreader:
image

But I was able to claim and the account was created successfully.

I tried to debugg this line: `at getAccountAssetsForSelectedAccount (getAccountAssetsForSelectedAccount.ts:16:1)` Does the error occur when it loads the dashboard? 

`getAssetFromPersistedAssets(COIN_TYPE[networkProtocol].toString())`

Maybe it's setting the wrong network protocol for some reason?

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
Looking great 💪🏼 
Just 2 comments:
- I would flip the buttons in this screen 
image
- "rerun" button doesnt respond, repro steps:
```
use the same seed in desktop & mobile, a seed that has unclaimed funds
get into the screen where your next step is just claiming, on both desktop and mobile
claim with desktop, so mobile cant claim anymore, though it knowns about the transaction
if will display an error because it cant claim twice the same funds
---- until here, all is good behavior
firefly mobile suggests "rerun", you press the button, nothing happens
```
image

**Review by begonaalvarezd — APPROVED** [review]
🚀

**Review by maxwellmattryan — APPROVED** [review]

**Pull request merged**